### DISCUSSION TITLE
Diamond dependencies checked twice

### DISCUSSION THREAD

**Discussion by ssbarnea** [go-task/task#756] [Q&A] [answered]
Apparently if I have diamond dependencies they are checked twice, thus polluting the tool output. I happen to have a `setup` step that is basically supposed to install dev tools and thus is a direct or indirect dependency of almost any other task.

For this reason, I seen taskfile displaying it as checked multiple times. Performance wise is not a big deal as that is skipped, but it does affect the visual, due to extra output.

Any idea if we can avoid displaying these multiple times?

**Comment by andreynering** [comment, marked as answer]
Declaring `run: once` for the task is probably the solution:

https://taskfile.dev/usage/#limiting-when-tasks-run

You can also set at the root of the file if you want that by default for all tasks.

**Reply by ssbarnea** [reply]
So `run: once` refers to per task invocation and you already considered a default for it. Awesome!

Now that I seen the documentation example, i do also understand why that was not the default. Thanks.

**Reply by ghostsquad** [reply]
Anyone else think `run: when_changed` should be the default? Not that we can actually make it the default (backwards incompatible). Just curious how many people would find it useful? Similarly, I'm curious how many people have diamond dependencies. 🤔

**Reply by andreynering** [reply]
In theory we could change the default in v4.
